service.upnext/resources/lib/player.py

```diff
@@ -0,0 +1,424 @@
+import xbmc
+import resources.lib.utils as utils
+from resources.lib.upnext import UpNext
+from resources.lib.stillwatching import StillWatching
+import sys
+
+if sys.version_info < (2, 7):
+    import simplejson as json
+else:
+    import json
+
+# service class for playback monitoring
+class Player(xbmc.Player):
+    # Borg - multiple instances, shared state
+    _shared_state = {}
+
+    xbmcplayer = xbmc.Player()
+    logLevel = 0
+    currenttvshowid = None
+    currentepisodeid = None
+    playedinarow = 1
+
+    def __init__(self, *args):
+        self.__dict__ = self._shared_state
+        self.logMsg("Starting playback monitor service", 1)
+        self.setupFromSettings()
+        xbmc.Player.__init__(self)
+
+    def logMsg(self, msg, lvl=1):
+        self.className = self.__class__.__name__
+        utils.logMsg("%s %s" % (utils.addon_name(), self.className), msg, int(lvl))
+
+    def onPlayBackStarted(self):
+        # Will be called when kodi starts playing a file
+        self.addon_data = {}
+        if utils.settings("developerMode") == "true":
+            self.developerPlayPlayback()
+
+    def getNowPlaying(self):
```

Comment:
It looks odd to use the json api to get the video InfoTag in a Player class instance